packages/snap-preact-components/src/components/Layouts/utils/Container/Container.tsx

```diff
@@ -0,0 +1,246 @@
+				textAlign,
+				flexBasis,
+				flexGrow,
+				flexShrink,
+				flexWrap,
+				flexFlow,
+				gap,
+				rowGap,
+				columnGap,
+				item,
+				justifyContent,
+				alignItems,
+				alignContent,
+				alignSelf,
+				width,
+				height,
+			}),
+			style,
+		];
+	} else if (style) {
+		styling.css = [style];
+	}
+
+	return (
+		<CacheProvider>
+			{href ? (
+				<a {...styling} href={href} className={classnames(item ? 'ss__container-item' : 'ss__container', className)}>
+					{children}
+				</a>
+			) : (
+				<div {...styling} className={classnames(item ? 'ss__container-item' : 'ss__container', className)}>
+					{children}
+				</div>
+			)}
+		</CacheProvider>
+	);
+});
+
+export interface ContainerProps extends ComponentProps {
+	children?: ComponentChildren;
+	width?: string | number;
+	height?: string | number;
+	alignItems?:
+		| 'normal'
+		| 'stretch'
+		| 'center'
+		| 'start'
+		| 'end'
+		| 'flex-start'
+		| 'flex-end'
+		| 'self-start'
+		| 'self-end'
+		| 'baseline'
+		| 'first baseline'
+		| 'last baseline'
+		| 'safe center'
+		| 'unsafe center'
+		| 'inherit'
+		| 'initial'
+		| 'revert'
+		| 'revert-layer'
+		| 'unset';
+	alignContent?:
+		| 'center'
+		| 'start'
+		| 'end'
+		| 'flex-start'
+		| 'flex-end'
+		| 'normal'
+		| 'baseline'
+		| 'first baseline'
+		| 'last baseline'
+		| 'space-between'
+		| 'space-around'
+		| 'space-evenly'
+		| 'stretch'
+		| 'safe center'
+		| 'unsafe center'
+		| 'inherit'
+		| 'initial'
+		| 'revert'
+		| 'revert-layer'
+		| 'unset';
+	flexWrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
+	flexDirection?: 'row' | 'row-reverse' | 'column' | 'column-reverse';
+	flexGrow?: number | 'inherit' | 'initial' | 'revert' | 'revert-layer' | 'unset';
+	flexShrink?: number | 'inherit' | 'initial' | 'revert' | 'revert-layer' | 'unset';
+	alignSelf?:
+		| 'auto'
+		| 'normal'
+		| 'center'
+		| 'start'
+		| 'end'
+		| 'self-start'
+		| 'self-end'
+		| 'flex-start'
+		| 'flex-end'
+		| 'baseline'
+		| 'first baseline'
+		| 'last baseline'
+		| 'stretch'
+		| 'safe center'
+		| 'unsafe center'
+		| 'inherit'
+		| 'initial'
+		| 'revert'
+		| 'revert-layer'
+		| 'unset';
+	flexFlow?:
+		| 'row'
+		| 'row-reverse'
+		| 'column'
+		| 'column-reverse'
+		| 'nowrap'
+		| 'wrap'
+		| 'wrap-reverse'
+		| 'row nowrap'
+		| 'column wrap'
+		| 'column-reverse wrap-reverse'
+		| 'inherit'
+		| 'initial'
+		| 'revert'
+		| 'revert-layer'
+		| 'unset';
+	justifyContent?:
+		| 'center'
+		| 'start'
+		| 'end'
+		| 'flex-start'
+		| 'flex-end'
+		| 'left'
+		| 'right'
+		| 'normal'
+		| 'space-between'
+		| 'space-around'
+		| 'space-evenly'
+		| 'stretch'
+		| 'safe center'
+		| 'unsafe center'
+		| 'inherit'
+		| 'initial'
+		| 'revert'
+		| 'revert-layer'
+		| 'unset';
+	flex?: string | number;
+	flexBasis?: string | number;
+	textAlign?: 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit';
+	gap?: string | number;
+	rowGap?: string | number;
+	columnGap?: string | number;
+	item?: boolean;
+	href?: string;
```

Comment:
Since this is a clickable thing that clients may want to tie into, do you think we should add an `onClick` prop?

Reply:
Wouldn't that open the door for some unwanted crazy customization?

Reply:
Currently the Button component is needed to do `onClick` functionality. My thinking here was that having a link may also require some additional tracking, and there is no possible way to handle that currently (outside of the Button component).

Reply:
Let's leave it off for now until use case arises.